### PR TITLE
DCOS-39895: Region Information in Services Table

### PR DESCRIPTION
**Pull request by juliangieseke**
## Feature Branch

This is a feature branch and code already was reviewed, this PR has to be checked for overall functionality

## Testing

when this is done there are region information on services table

## Screenshot
screen shot 2018-08-24 at 12 48 02
(only region column in this PR)

### CODE REVIEW

File: plugins/services/src/js/containers/services/ServicesTable.js [hunk 1]
```diff
@@ -242,6 +245,30 @@ class ServicesTable extends React.Component {
...
+  renderRegions(prop, service) {
+    const localRegion = CompositeState.getMasterNode().getRegionName();
+    let regions = service.getRegions();
+
+    regions = regions.map(
+      region => (region === localRegion ? region + " (Local)" : region)
+    );
+
+    if (regions.length === 0) {
+      regions.push("N/A");
```
Comment (weblancaster) [inline plugins/services/src/js/containers/services/ServicesTable.js]: We are starting to have this `N/A` all over the code base, I would rather have a stateless component that renders that. I also record that at some point last year we wanted to replace `N/A` to `-` and I believe that we created a helper or a component that returns `-`

File: plugins/services/src/js/structs/Service.js [hunk 2]
```diff
@@ -7,6 +7,10 @@ import ServiceSpec from "./ServiceSpec";
 import VolumeList from "./VolumeList";
 
 module.exports = class Service extends Item {
+  constructor() {
+    super(...arguments);
+    this._regions = undefined;
```
Comment (weblancaster) [inline plugins/services/src/js/structs/Service.js]: why `undefined` and not `null`?

File: plugins/services/src/js/structs/__tests__/PodInstance-test.js [hunk 3]
```diff
@@ -52,6 +52,32 @@ describe("PodInstance", function() {
     });
   });
 
+  describe("#getAgentRegion", function() {
+    it("returns the correct value", function() {
```
Comment (weblancaster) [inline plugins/services/src/js/structs/__tests__/PodInstance-test.js]: Can we have a more descriptive title for all tests, please? e.g https://github.com/dcos/dcos-ui/pull/3154/files#diff-898ccecc5fbc3897c2529139b461bcccR78

File: plugins/services/src/js/containers/services/ServicesTable.js (hunk 1 above)
Comment (mperrotti) [reply plugins/services/src/js/containers/services/ServicesTable.js]: There was also a recent case where the design team preferred having a blank cell. Should we talk to design about how we decide when to use `N/A` vs `-` vs nothing?

Comment (weblancaster) [reply plugins/services/src/js/containers/services/ServicesTable.js]: @mperrotti good point. I remember that case too now which was for services version.

Comment (juliangieseke) [reply plugins/services/src/js/containers/services/ServicesTable.js]: Yeah, I like the Idea of having small `<Empty />` and `<NotAvailable />` components to keep things aligned. For the empty case we also switch between `null`, `<div />` and `<span />` 😕

However I'd say it is not in scope for this feature branch to create these components, we can clean up the code later - @weblancaster @mperrotti can one of you create Epics and facilitate them? 🙇

File: plugins/services/src/js/structs/Service.js (hunk 2 above)
Comment (juliangieseke) [reply plugins/services/src/js/structs/Service.js]: why `null` and not `undefined`? :trollface: 

scnr 😅  - havent thought about it when writing the code 🤷‍♂️ I'm not aware that we use one of them consistently thorough our codebase. Do we want to do so?

Comment (juliangieseke) [reply plugins/services/src/js/structs/Service.js]: thinking about it now, an empty array would have been better to be typesafe 🤔 will change 👍

**Comment by juliangieseke** [comment]
rephrased test description and rebased branch. /cc @GeorgiSTodorov 🙇

**Pull request merged**